assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+            $scope.options.pickedPushToImageStream = $scope.pushTo.imageStreams[0];
+            $scope.clearSelectedOutputTag();
+            // If defined Image Stream is not present in the Image Streams of picked Namespace set tag to empty string,
+            // so the user has to pick from a existing Image Streams.
+          } else if (!$scope.pushTo.imageStreams.contains($scope.options.pickedPushToImageStream)) {
+            $scope.options.pickedPushToImageStreamTag = "";
+          }
+        } else {
+          $scope.options.pickedPushToImageStream = "";
+          $scope.options.pickedPushToImageStreamTag = "";
+        }
+      });
+    }
+
+    $scope.clearSelectedOutputTag = function() {
+      var tags = $scope.pushTo.tags[$scope.options.pickedPushToImageStream];
+      $scope.options.pickedPushToImageStreamTag = _.find(tags, function(tag) { return tag == "latest" }) || tags[0] || "latest";
+    }
+
+    $scope.save = function() {
+      $scope.disableInputs = true;
+      // Update Configuration
+      $filter('buildStrategy')($scope.updatedBuildConfig).forcePull = $scope.options.forcePull;
+      if ($scope.strategyType === "Docker") {
+        $filter('buildStrategy')($scope.updatedBuildConfig).noCache = $scope.options.noCache;
+      }
+
+      // If binarySource check if the AsFile string is set and construct the object accordingly.
+      if ($scope.options.binaryAsFile) {
+        if ($scope.options.binaryAsFile !== "") {
+          $scope.updatedBuildConfig.spec.source.binary.asFile = $scope.options.binaryAsFile;
+        } else {
+          $scope.updatedBuildConfig.spec.source.binary = {};
+        }
+      }
+
+      // If imageSource is present update From and Paths.
+      if ($scope.sources.image) {
+        var updatedImageSourcePath = [];
+        angular.forEach($scope.imageSourcePaths, function(v, k) {
+          var env = {
+            sourcePath: k,
+            destinationDir: v
+          };
+          updatedImageSourcePath.push(env);
+        });
+        $scope.updatedBuildConfig.spec.source.image.paths  = updatedImageSourcePath;
+
+        // Construct updated imageSource builder image object based on it's kind
+        var from = {};
+        if ($scope.options.pickedImageSourceType === "ImageStreamTag") {
+          from = {
+            kind: $scope.options.pickedImageSourceType,
+            namespace: $scope.options.pickedImageSourceNamespace,
+            name: $scope.options.pickedImageSourceImageStream + ":" + $scope.options.pickedImageSourceImageStreamTag
+          };
+        } else if ($scope.options.pickedImageSourceType === "DockerImage") {
+          from = {
+            kind: $scope.options.pickedImageSourceType,
+            name: $scope.options.pickedImageSourceDockerImage
+          };
+        }
+        $scope.updatedBuildConfig.spec.source.image.from = from;
+      }
+
+      // Construct updated builder image object based on it's kind
+      var from = {};
```

Comment:
`var` is function scoped... don't know if it will consistently complain, but prob should declare `from` only once in the function.